### ISSUE TITLE
Expo template (JS) issues

### ISSUE THREAD

**Issue by mangkoran** [realm/realm-js#4161]
### How frequently does the bug occur?

All the time

### Description

I followed the instruction from the recently merged https://github.com/realm/realm-js/pull/4145:
```sh
npx react-native init AwesomeRealmProject --template @realm/expo-template-js
```
However, I stumbled upon multiple issues:

1. The command won't run `yarn install` somehow.
```sh
✖ Installing dependencies
error Error: Command failed: yarn install
```
This issue easily workaround-ed by manually run `yarn install` or `npm install`

2. Whenever I run `expo start` and open the app in Android, it only shows white blank screen even though there is no error (the bundling completed successfully)

![image](https://user-images.githubusercontent.com/57625126/146681774-27f06cec-0a34-4f51-836a-81e29caa286c.png)

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

1. Run `npx react-native init AwesomeRealmProject --template @realm/expo-template-js`, as stated in https://github.com/realm/realm-js/blob/643fff6f147685a5494a8ddc6bd4759061aae9af/packages/expo-template-js/README.md
2. You will get the first issue
3. `cd` to the bootstrapped project
4. Manually solve first issue with either `yarn install` or `npm install` (I tried both, but most of my project use `npm`)
5. Run `expo start`, then `open Android`
6. You will get the second issue

### Version

10.10.1

### What SDK flavour are you using?

MongoDB Realm (i.e. Sync, auth, functions)

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

WSL2 in Windows 11 22000.376

### Build environment

Which debugger for React Native: I am not sure
Package:
- node: v14.18.1
- expo-cli: v5.0.1

Android device: Windows Subsystem for Android
Note: For build environment, I fully use WSL2, so you may consider that I use Linux. For WSA, I have other React Native/Expo project too that run successfully without any issue.

Related issue: https://github.com/realm/realm-js/issues/4142

**Comment by takameyer** [comment]
Thanks for trying out the template, and sorry that it's not working smoothly! This is using the 'expo-dev-client', so I would recommend using the two commands 'yarn android' and 'expo start --dev-client'.  Ill make some updates to the package.json and documentation in the near future to make this clearer.

**Comment by mangkoran** [comment]
Thanks for the quick response! However, after tried `yarn android`, I got this error message:
```
~/repos/AwesomeRealmProject via  v14.18.1
❯ yarn android
yarn run v1.22.15
$ expo run:android
› Building app...

FAILURE: Build failed with an exception.

* What went wrong:
Could not open settings generic class cache for settings file '/home/apiqnzr/repos/AwesomeRealmProject/android/settings.gradle' (/home/apiqnzr/.gradle/caches/6.7/scripts/bg08yym9nro6faonxzfqk72j1).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 61

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 470ms
/home/apiqnzr/repos/AwesomeRealmProject/android/gradlew exited with non-zero code: 1
Error: /home/apiqnzr/repos/AwesomeRealmProject/android/gradlew exited with non-zero code: 1
    at ChildProcess.completionListener (/usr/lib/node_modules/expo-cli/node_modules/@expo/spawn-async/src/spawnAsync.ts:65:13)
    at Object.onceWrapper (events.js:520:26)
    at ChildProcess.emit (events.js:400:28)
    at maybeClose (internal/child_process.js:1058:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
    ...
    at spawnAsync (/usr/lib/node_modules/expo-cli/node_modules/@expo/spawn-async/src/spawnAsync.ts:26:19)
    at spawnGradleAsync (/usr/lib/node_modules/expo-cli/src/commands/run/android/spawnGradleAsync.ts:83:18)
    at assembleAsync (/usr/lib/node_modules/expo-cli/src/commands/run/android/spawnGradleAsync.ts:57:16)
    at actionAsync (/usr/lib/node_modules/expo-cli/src/commands/run/android/runAndroid.ts:143:9)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
I don't really familiar with `gradle`, but after searching Google a bit, it seems that it might be caused with incompatible JDK? I am using `java-17` tho, and the template seems it uses `gradle` 6.7.

**Comment by mangkoran** [comment]
After tinkering with the template a bit, I managed to run the template

![image](https://user-images.githubusercontent.com/57625126/146708781-ab787644-c812-4bfa-8c07-fd3745ea63f9.png)

However, I agree with your comment:
>  Ill make some updates to the package.json and documentation in the near future to make this clearer.

The template definitely needs more verbose step-by-step tutorial. As for me, this is my first time running project with `gradle`, Android SDK and such.

Also, we also need documentation to use the Expo-Realm in existing project.

For the template, if you are going to improve the documentation, feel free to HMU!

**Comment by takameyer** [comment]
I'll also add a section about (EAS)[https://docs.expo.dev/build/introduction/].  Basically that does the build in the cloud and makes it downloadable to your device.  With that you can connect to the dev-client to debug your app, and avoid using Gradle or Android Studio.

**Comment by mangkoran** [comment]
Definitely, EAS will help greatly in production!

For the time being, I have several questions regarding Expo-Realm integration:

1. As far as I have read the steps you mentioned in #4142, it seems that we are required to have a project bootstraped with `react-native` CLI. Is this correct?

**Comment by takameyer** [comment]
> As far as I have read the steps you have mentioned in Expo Template for Realm #4142, it seems that we are required to have a project bootstraped with react-native CLI before we can proceed to integrate them with Expo-Realm. Is this correct?

At this moment, yes. But that's a good point, we should be making it possible to initialize the template with `expo init -t`, which doesn't work at this time with the current template.

> Could you link some of reference you have used on how to integrate Expo with third-party library?
Since this is using the `expo-dev-client`, you will be able to just follow the installation instructions of the third-party library you want to add.  Afterwards you must rebuild either through EAS or by building locally with the `run` commands.

**Comment by takameyer** [comment]
Im also in contact with Expo and they have shared some ideas how to make the templates more developer friendly. There will be some updates in the near future

**Comment by takameyer** [comment]
@mankoran I've made significant updates to the templates as well as the documentation.  They are now installable with `expo init -i`.  They also make use of our react hooks library `@realm/react`.  If you have any problems, feel free to submit a new issue :)

**Comment by mangkoran** [comment]
Thank you for your hard work! After looked into the new templates, I noticed some typo there. Just small issue, but I will open a PR for it!

**Comment by lkoehl** [comment]
> They also make use of our react hooks library `@realm/react`.
Is there an example to use the hooks with a sync configuration?

**Comment by takameyer** [comment]
We are working on some documentation.  In the meantime, something like this would work:
```
const [user, setUser] = useState()

return (
<>
{user?
<RealmProvider sync={{user, partition: "some partition value"}}>
 <StartScreen/>
</RealmProvider>
:
<LoginScreen setUser={setUser}/>
}
</>
)
```
